dotnet/src/Connectors/Connectors.MicrosoftML/TextEmbedding/MicrosoftMLTextEmbeddingGeneration.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+#pragma warning disable IDE0005
+using Microsoft.ML.Tokenizers;
+#pragma warning restore IDE0005
+
+using Microsoft.SemanticKernel.AI.Embeddings;
+
+#pragma warning disable IDE0130
+// ReSharper disable once CheckNamespace - Using NS of KernelConfig
+namespace Microsoft.SemanticKernel;
+#pragma warning restore IDE0130
+/// <summary>
+/// Provides text embedding generation using a BPE tokenizer.
+/// </summary>
+public sealed class MicrosoftMLTextEmbeddingGeneration : ITextEmbeddingGeneration
+{
+    private string vocabFilePath;
+    private string mergeFilePath;
+    private Tokenizer tokenizer;
+    /// <summary>
+    /// Gets the logger instance.
+    /// </summary>
+    private ILogger Logger { get; set; }
+
+    /// <summary>
+    /// Initializes a new instance of the MicrosoftMLTextEmbeddingGeneration class.
+    /// </summary>
+    /// <param name="vocabFilePath">The JSON file path containing the dictionary of string keys and their ids.</param>
+    /// <param name="mergeFilePath">The file path containing the tokens's pairs list.</param>
+    /// <param name="loggerFactory">Optional logger factory for logging.</param>
+    public MicrosoftMLTextEmbeddingGeneration(
+        string vocabFilePath,
+        string mergeFilePath,
+        ILoggerFactory? loggerFactory = null
+    )
+    {
+        this.vocabFilePath = vocabFilePath;
+        this.mergeFilePath = mergeFilePath;
+        this.tokenizer = new Tokenizer(new Bpe(vocabFilePath, mergeFilePath));
```

Comment:
I wonder if it'd be more flexible to just supply a Tokenizer object so consumers could supply a RobertaPreTokenizer, etc.

Reply:
I went ahead and took a stab at adding this. Highlights that there is issue with embeddings calculations.